runelite-client/src/main/java/net/runelite/client/plugins/chatfilter/ChatFilterPlugin.java

```diff
@@ -383,4 +395,63 @@ boolean shouldFilterByName(final String playerName)
 		}
 		return false;
 	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.rightClickFilter())
+		{
+			return;
+		}
+
+		int type = event.getType();
+
+		if (type >= MENU_ACTION_DEPRIORITIZE_OFFSET)
+		{
+			type -= MENU_ACTION_DEPRIORITIZE_OFFSET;
+		}
+		final MenuAction menuAction = MenuAction.of(type);
+
+		if (menuAction.equals(RUNELITE_PLAYER))
+		{
+			MenuEntry[] menuEntries = client.getMenuEntries();
+			menuEntries = Arrays.copyOf(menuEntries, menuEntries.length + 1);
+
+			final String playerName = Text.standardize(event.getTarget().split("\\s+")[0]);
+			final boolean nameFiltered = shouldFilterByName(playerName);
+
+			final MenuEntry filterEntry = menuEntries[menuEntries.length - 1] = new MenuEntry();
+			filterEntry.setOption(nameFiltered ? REMOVE_FILTER : FILTER_CHAT);
+			filterEntry.setTarget(event.getTarget());
+			filterEntry.setType(MenuAction.RUNELITE.getId());
+
+			client.setMenuEntries(menuEntries);
+		}
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked click)
+	{
+		if (click.getMenuAction() != MenuAction.RUNELITE ||
+			!(click.getMenuOption().equals(FILTER_CHAT) || click.getMenuOption().equals(REMOVE_FILTER)))
+		{
+			return;
+		}
+
+		final String playerName = Text.standardize(click.getMenuTarget().split("\\s+")[0]);
+		String filteredNames = configManager.getConfiguration("chatfilter", "filteredNames");
+
+
+		if (click.getMenuOption().equals(FILTER_CHAT))
+		{
+			config.setNamesToFilter(filteredNames.isEmpty() ? playerName : filteredNames + "\n" + playerName);
+		}
+		else
+		{
+			filteredNames = filteredNames.replaceAll(playerName, "");
```

Comment:
A newline splitter is defined at the top of this plugin class, use that to fetch filtered names  and filter entries exactly matching the clicked player's name instead of using `replaceAll`. This would, for instance, break other name filters if you tried to unfilter the name of a player with a one-letter name.